suc/pkg/service/config.go

```diff
@@ -0,0 +1,102 @@
+package service
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/rancher/wins/cmd/server/config"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	ConfigDirEnvVar      = "CATTLE_WINS_CONFIG_DIR"
+	DebugEnvVar          = "CATTLE_WINS_DEBUG"
+	AgentStringTLSEnvVar = "STRICT_VERIFY"
+)
+
+const (
+	defaultConfigFile = "c:/etc/rancher/wins/config"
+)
+
+func getConfigPath(path string) string {
+	configDirEnv := os.Getenv(ConfigDirEnvVar)
+	if path == "" && configDirEnv != "" {
+		path = configDirEnv
+	}
+	if path == "" {
+		path = defaultConfigFile
+	}
+	return path
+}
+
+// loadConfig utilizes the ConfigDirEnvVar environment variable and path parameter to load
+// a config file located on the host. If both are provided, the path parameter will take
+// precedence. If neither are provided, then the defaultConfigFile path is used.
+func loadConfig(path string) (*config.Config, error) {
+	cfg := config.DefaultConfig()
+	err := config.LoadConfig(getConfigPath(path), cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+// loadConfig utilizes the ConfigDirEnvVar environment variable and path parameter to save
+// a config file on the host. If both are provided, the path parameter will take
+// precedence. If neither are provided, then the defaultConfigFile path is used.
+func saveConfig(cfg *config.Config, path string) error {
+	return config.SaveConfig(getConfigPath(path), cfg)
+}
+
+// UpdateConfigFromEnvVars is responsible for updating the rancher-wins config file
+// based off of the presence of particular environment variables. The path parameter is used
+// to specify the location of the config file on the host. If path is left empty, the defaultConfigFile
+// constant is used. The config file will only be updated if a given environment variable is present, and its
+// value does not equal the currently set value in the config file. UpdateConfigFromEnvVars returns a boolean
+// indicating if the config file has been updated and any errors encountered.
+func UpdateConfigFromEnvVars(path string) (bool, error) {
+	logrus.Info("Loading config from host")
+	cfg, err := loadConfig(path)
+	if err != nil {
+		return false, fmt.Errorf("failed to load config: %v", err)
+	}
+
+	configNeedsUpdate := false
+	logrus.Infof("Checking the %s value. This is a boolean flag, expecting 'true' or 'false'", DebugEnvVar)
+	if v := os.Getenv(DebugEnvVar); v != "" {
+		logrus.Infof("Found value '%s' for %s", v, DebugEnvVar)
+		givenBool := strings.ToLower(DebugEnvVar) == "true"
+		if cfg.Debug != givenBool {
+			cfg.Debug = givenBool
+			configNeedsUpdate = true
+		}
+	}
+
+	logrus.Infof("Checking the %s value. This is a boolean flag, expecting 'true' or 'false'", AgentStringTLSEnvVar)
+	if v := os.Getenv(AgentStringTLSEnvVar); v != "" {
+		logrus.Infof("Found value '%s' for %s", v, AgentStringTLSEnvVar)
+		tlsMode, err := strconv.ParseBool(strings.ToLower(v))
```

Comment:
If the expected value is "true" or "false", it is better to use the same string-compare method as the above section to avoid falsely parsing other invalid values.